clang/lib/Sema/SemaInit.cpp

```diff
@@ -10043,12 +10043,19 @@ QualType Sema::DeduceTemplateSpecializationFromInitializer(
     //   When [...] the constructor [...] is a candidate by
     //    - [over.match.copy] (in all cases)
     if (TD) {
-      SmallVector<Expr *, 8> TmpInits;
-      for (Expr *E : Inits)
+
+      // As template candidates are not deduced immediately,
+      // persist the arry in the overload set.
```

Comment:
```suggestion
      // persist the array in the overload set.
```

Reply:
Copy-pasted in a whole bunch of places too!